src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.cpp

```diff
@@ -0,0 +1,248 @@
+
+  const auto deriv_log_r = [&]() {
+    auto l_deriv_log_r = make_with_value<tnsr::i<DataType, 3>>(x, 0.0);
+    for (size_t i = 0; i < 3; ++i) {
+      l_deriv_log_r.get(i) =
+          deriv_log_r_denom *
+          (x_minus_center.get(i) + gsl::at(spin_, i) * a_dot_x_over_rsquared);
+    }
+    return l_deriv_log_r;
+  }();
+
+  const DataType H_denom = 1.0 / (square(r_squared) + a_dot_x_squared);
+  const DataType H = mass_ * sqrt(r_squared) * r_squared * H_denom;
+
+  const auto deriv_H = [&]() {
+    auto l_deriv_H = make_with_value<tnsr::i<DataType, 3>>(x, 0.0);
+    const DataType fac1 = H * (3.0 - 4.0 * square(r_squared) * H_denom);
+    const DataType fac2 = H * (2.0 * H_denom * a_dot_x);
+    for (size_t i = 0; i < 3; ++i) {
+      l_deriv_H.get(i) = fac1 * deriv_log_r.get(i) - fac2 * gsl::at(spin_, i);
+    }
+    return l_deriv_H;
+  }();
+
+  const auto a_cross_x = [](const auto& a, const auto& coord) {
+    auto l_a_cross_x = make_with_value<tnsr::i<DataType, 3>>(coord, 0.0);
+    get<0>(l_a_cross_x) = a[1] * get<2>(coord) - a[2] * get<1>(coord);
+    get<1>(l_a_cross_x) = a[2] * get<0>(coord) - a[0] * get<2>(coord);
+    get<2>(l_a_cross_x) = a[0] * get<1>(coord) - a[1] * get<0>(coord);
+    return l_a_cross_x;
+  }(spin_, x_minus_center);
+
+  const DataType denom = 1.0 / (r_squared + a_squared);
+  const DataType r = sqrt(r_squared);
+
+  const auto null_form = [&]() {
+    auto l_null_form = make_with_value<tnsr::i<DataType, 3>>(x, 0.0);
+    const DataType fac = a_dot_x / r;
+    for (size_t i = 0; i < 3; ++i) {
+      l_null_form.get(i) = denom * (r * x_minus_center.get(i) -
+                                    a_cross_x.get(i) + fac * gsl::at(spin_, i));
+    }
+    return l_null_form;
+  }();
+
+  const auto deriv_null_form = [&]() {
+    auto l_deriv_null_form = make_with_value<tnsr::ij<DataType, 3>>(x, 0.0);
+    for (size_t i = 0; i < 3; i++) {
+      for (size_t j = 0; j < 3; j++) {
+        l_deriv_null_form.get(j, i) =
+            denom * (gsl::at(spin_, i) * gsl::at(spin_, j) / r +
+                     (x_minus_center.get(i) - 2.0 * r * null_form.get(i) -
+                      a_dot_x_over_rsquared * gsl::at(spin_, i)) *
+                         deriv_log_r.get(j) * r);
+        if (i == j) {
+          l_deriv_null_form.get(j, i) += denom * r;
+        } else {  //  add denom*epsilon^ijk a_k
+          size_t k = (j + 1) % 3;
+          if (k == i) {  // j+1 = i (cyclic), so choose minus sign
+            k++;
+            k = k % 3;  // and set k to be neither i nor j
+            l_deriv_null_form.get(j, i) -= denom * gsl::at(spin_, k);
+          } else {  // i+1 = j (cyclic), so choose plus sign
+            l_deriv_null_form.get(j, i) += denom * gsl::at(spin_, k);
+          }
+        }
+      }
+    }
+    return l_deriv_null_form;
+  }();
+
+  const constexpr double null_vector_0 = -1.0;
+  const DataType lapse_squared = 1.0 / (1.0 + 2.0 * H * square(null_vector_0));
```

Comment:
We're really confused by `null_vector_0`'s use on this line (`square`) If this changes for boosted solutions a comment would be useful